src/dds_manager.hpp

```diff
@@ -83,6 +83,13 @@ bool DDSManager::registerTopic(const std::string& topicName, const STD_QOS::QosT
     CORBA::String_var tn = ts->get_type_name();
     status = ts->register_type(m_domainParticipant, tn.in());
     checkStatus(status, "register_type");
+
+    // Make sure the topic actually registered. If not, there was an issue in the config file
+    if (m_domainParticipant == nullptr)
```

Comment:
Should this check be above register_type, since we're passing `m_domainParticipant` as an argument?

Reply:
Good idea.  I moved that check up toward the top of the function.